chaos/table_list_paging.go

```diff
@@ -0,0 +1,118 @@
+package chaos
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+type Item struct {
+	Id   string
+	Page int
+}
+
+type PagingResponse struct {
+	NextPage int
+}
+
+type ListResponse struct {
+	Items []Item
+	Resp  *PagingResponse
+}
+
+// paging is 0 based, ranging from 0 to 4
+var MaxPage = 5
+var noMorePages = -1
+
+// the number of times the function should fail/retry
+var failureCount = 200
+
+var errorAfterPages = 3
+var retryListError = map[string]int{}
+var listErrorString = "retriableError"
+var listMutex = &sync.Mutex{}
+
+func listPagingTable() *plugin.Table {
+	return &plugin.Table{
+		Name:             "chaos_list_paging",
+		Description:      "Chaos table to test the list function supporting pagination fails to send results after some pages with a non fatal error",
+		DefaultTransform: transform.FromCamel(),
+		List: &plugin.ListConfig{
+			Hydrate: listPagingFunction,
+		},
+
+		Columns: []*plugin.Column{
+			{Name: "id", Type: proto.ColumnType_STRING},
+			{Name: "page", Type: proto.ColumnType_INT},
+		},
+	}
+}
+
+func listPagingFunction(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	nextPage := 0
+
+	listPage := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
add comment: define function to fetch a page of data - we will pass this to the sdk RetryHydrate call